lib/features/interactive.js

```diff
@@ -0,0 +1,112 @@
+var debug = require('debug')('docker-worker:features:interactive');
+var DockerExecServer = require('docker-exec-websocket-server').DockerExecServer;
+var http = require('http');
+var https = require('https');
+var slugid = require('slugid');
+var Promise = require('promise');
+var url = require('url');
+var fs = require('fs');
+
+//Number of attempts to find a port before giving up
+const MAX_RANDOM_PORT_ATTEMPTS = 20;
+
+export default class WebsocketServer {
+  constructor () {
+    this.path = '/' + slugid.v4();
+  }
+
+  async started (task) {
+    debug('creating ws server');
+    var httpServ;
+    if (task.runtime.interactive.ssl) {
+      var readFile = Promise.denodeify(fs.readFile);
```

Comment:
we use 'mz' package elsewhere so it might be easy to just use that here instead of promisefying it.